### ISSUE TITLE
_BitScanReverse inWindow Oscillator

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#6726]
See #6721 

Can't we be a bit more portable than that?

**Comment by baconpaul** [comment]
Well the short answer to this is "no you really can't at least not until c++20" so moving this into currently unscheduled.

**Comment by mkruselj** [comment]
@baconpaul We're C++20 now so we could maybe push this into 1.4?

**Comment by baconpaul** [comment]
FunnyI think the c++ workaround we have for mac/lin uses the wrong builtin - it uses ctz not clz, where clz is the reverse scan equivalent. 

So we need to test this pretty carefully a/b windows to make sure we don't break it when we move to std::bit_width - 1 (which i think is the equivalent value).

**Comment by mkruselj** [comment]
Or it would be retaining the "wrong" builtin and going with `std::countr_zero()`, I guess.

EDIT: I git blamed this, and looks like the wrong builtin was there ever since Claes originally open sourced the code! So I guess we need to retain that behavioral difference on Win vs Mac, then?

**Comment by baconpaul** [comment]
right but only on mac/lin.

**Comment by baconpaul** [comment]
and yeah this is since epoch.

**Comment by mkruselj** [comment]
Oof, but that difference basically completely breaks mipmapping on non-Windows. That is very very bad... We ought to fix that, after all, I reckon...

**Comment by baconpaul** [comment]
yeah
you can try it both ways on windows to hear diff

**Comment by mkruselj** [comment]
Yeah, it's horrible once you start sweeping that Formant parameter towards its maximum!

We gotta fix this even if it's a breaking change on *nixes. We shouldn't have platform-dependent sound differences.

**Comment by baconpaul** [comment]
it does explain why i never really liked the window oscillator tho

**Comment by baconpaul** [comment]
and agree with fixing it
